app/operations/candidates/apply.rb

```diff
@@ -54,7 +54,16 @@ def call
         }
       ).call
 
+      candidate
+    end
+
+    # File uploading did not work correctly inside the main transaction.
+    case Candidates::UploadFile.new(candidate:, actor_account:, file:, cv: true).call
+    in Success(_)
       Success()
+    in Failure[:file_invalid, e]
+      candidate.destroy!
+      Failure[:file_invalid, e]
```

Comment:
[discussion](https://toughbyte.slack.com/archives/C066V39GZPT/p1730798864887389?thread_ts=1730527148.486689&cid=C066V39GZPT) where agreed to move file uploading out from the main transaction and imitate rollback if file was not uploaded